src/v/kafka/server/tests/s3_imposter_fixture.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+#include "ssx/sformat.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/http/httpd.hh>
+
+#include <chrono>
+#include <exception>
+#include <map>
+#include <vector>
+
+/// Emulates S3 REST API for testing purposes.
+/// The imposter is a simple KV-store that contains a set of expectations.
+/// Expectations are accessible by url via GET, PUT, and DELETE http calls.
+/// Expectations are provided before impster starts to listen. They have
+/// two field - url and optional body. If body is set to nullopt, attemtp
+/// to read it using GET or delete it using DELETE requests will trigger an
+/// http response with error code 404 and xml formatted error message.
+/// If the body of the expectation is set by the user or PUT request it can
+/// be retrieved using the GET request or deleted using the DELETE request.
+class s3_imposter_fixture {
```

Comment:
I think that this is a third place in the coffee bar where the S3 imposter was used. @abhijat added http test helpers lib to the tree. So, I think that we can merge this but please add a tech-debt issue to GitHub.